scalafix-interfaces/src/main/java/scalafix/internal/interfaces/ScalafixInterfacesClassloader.java

```diff
@@ -0,0 +1,28 @@
+package scalafix.internal.interfaces;
+
+/**
+ * A classloader that shares only scalafix-interfaces classes from the parent classloader.
+ *
+ * This classloader is intended to be used as a parent when class-loading scalafix-cli.
+ * By using this classloader as a parent, it's possible to cast runtime instances from
+ * the scalafix-cli classloader into `scalafix.interfaces.Scalafix` from this classlaoder.
+ */
+public class ScalafixInterfacesClassloader extends ClassLoader {
+    private final ClassLoader parent;
+
+    public ScalafixInterfacesClassloader(ClassLoader parent) {
+        super(null);
+        this.parent = parent;
+    }
+
+    @Override
+    protected Class<?> findClass(String name) throws ClassNotFoundException {
+        if (name.startsWith("scalafix.interfaces")
+                // include types in scalafix.interfaces.* signatures
+                || name.startsWith("coursierapi")) {
```

Comment:
I initially had the new overloads in `ScalafixArguments` implemented within `scalafix-interfaces` as Java8 default methods (leaving `scalafix-cli` untouched), so this was not required. It did feel weird though, so I ended up pushing them down to `scalafix-interfaces`, keeping most of the logic in `scalafix-interfaces`. I am not sure what's best...

Reply:
This looks good